server/src/controllers/Blogs.js

```diff
@@ -2,61 +2,67 @@ const User = require("../models/User");
 const Blog = require("../models/Blog");
 
 module.exports = {
-  async allBlogs(_req, res) {
-    const blogs = await Blog.find();
-    res.status(200).json(blogs);
+
+  async allBlogs(_req, res, next) {
+    try{
+      const blogs = await Blog.find();
+      return res.status(200).json(blogs);
+    }catch(e){
+      return res.status(400).json({error:e.message});
+    }
   },
 
   async viewBlogById(req, res) {
-    const blogId = req.params.id;
-    const blog = await Blog.findById(blogId);
-    if (blog) {
-      res.status(200).json(blog);
-    } else {
-      res.status(404).json({
-        error: "The requested blog doesn't exist",
-      });
+    try{
+      const blogId = req.params.id;
+      const blog = await Blog.findById(blogId);
+      if (blog) {
+        return res.status(200).json(blog);
+      } else {
+        return res.status(404).json({
+          error: "The requested blog doesn't exist",
+        });
+      }
+    }catch(e){
+      return res.status(500).json({error:e.message})
     }
   },
 
-  async uploadBlog(req, res) {
-    // TODO: add isBlogAuthorized middleware
+  async uploadBlog(req, res, next) {
     try {
       // Assumed that req.body already has required fields
       const blog = await Blog.create(req.body);
-      res.status(201).json({
+      return res.status(201).json({
         id: blog._id,
       });
     } catch (error) {
       // console.log(error);
-      res.status(500).json({
-        error: error,
-      });
+      return res.status(500).json({message:error.message});
```

Comment:
Since 500 is a server side error, it'd be great if you could replace the key with "err" or "error" to signify that this is, in fact, an error

Reply:
Noted